### ISSUE TITLE
Compact string with precomputed hash

### ISSUE THREAD

**Issue by Boshen** [oxc-project/backlog#60]
> We are commonly hashing Atoms when e.g. building scope tree. Possibly could avoid repeated hashing and memory accesses by storing hash inline in Atom.

Rolldown wants this.

**Comment by rzvxa** [comment]
Isn't this just a duplicate of https://github.com/oxc-project/backlog/issues/46#issuecomment-2200571540

**Comment by Boshen** [comment]
Yes, that issue is for a overhaul of the Atom type, where this is just a wrapped type, which should be done separately as two tasks.

**Comment by rzvxa** [comment]
Yeah sorry for the comment, I came to this conclusion after sending that comment so that's why I marked it as resolved.

**Comment by overlookmotel** [comment]
Related to #46.

**Comment by overlookmotel** [comment]
I think #46 covers this. We're not really using `CompactStr` any more.
